packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -5,13 +5,16 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
+import 'package:flutter/cupertino.dart';
 
 import 'button.dart';
 import 'button_theme.dart';
 import 'material_button.dart';
 import 'theme.dart';
 import 'theme_data.dart';
 
+enum _FlatButtonType { material, adaptive }
+
 /// A material design "flat button".
 ///
 /// ### This class is obsolete, please use [TextButton] instead.
```

Comment:
I just realized this class is deprecated :S 

Let's add the adaptive constructor to TextButton as well.